contracts/marketplace/src/execute.rs

```diff
@@ -382,6 +399,61 @@ pub fn execute_accept_bid(
     Ok(res.add_event(event))
 }
 
+/// Migrates bids from the old index contract to the new index
+pub fn execute_migrate_bids(
+    deps: DepsMut,
+    _env: Env,
+    info: MessageInfo,
+    limit: u32,
+) -> Result<Response, ContractError> {
+    nonpayable(&info)?;
+
+    let sudo_params = SUDO_PARAMS.load(deps.storage)?;
+
+    ensure_eq!(
+        info.sender,
+        sudo_params.operator,
+        ContractError::Unauthorized {}
+    );
+
+    let old_bids = legacy_bids()
```

Comment:
Is it possible for this iteration to go over the block gas limit?

Reply:
the number of bids to process is an arg that is passed in, so would just need to lower that value if the transactions go over the gas limit